miio/fan_miot.py

```diff
@@ -525,3 +548,260 @@ def delay_off(self, minutes: int):
             raise FanException("Invalid value for a delayed turn off: %s" % minutes)
 
         return self.set_property("power_off_time", minutes)
+
+
+class OperationModeFanZA5(enum.Enum):
+    Nature = 0
+    Normal = 1
+
+
+class FanStatusZA5(DeviceStatus):
+    """Container for status reports for FanZA5."""
+
+    def __init__(self, data: Dict[str, Any]) -> None:
+        self.data = data
+
+    # TODO: Docstrings.
+    @property
+    def anion(self) -> bool:
+        return self.data["anion"]
+
+    @property
+    def battery_supported(self) -> bool:
+        return self.data["battery_supported"]
+
+    @property
+    def buttons_pressed(self) -> str:
+        code = self.data["buttons_pressed"]
+        if code == 0:
+            return "None"
+        if code == 1:
+            return "Power"
+        if code == 2:
+            return "Swing"
+        return "Unknown"
+
+    @property
+    def buzzer(self) -> bool:
+        return self.data["buzzer"]
+
+    @property
+    def child_lock(self) -> bool:
+        return self.data["child_lock"]
+
+    @property
+    def fan_level(self) -> int:
+        return self.data["fan_level"]
+
+    @property
+    def fan_speed(self) -> int:
+        return self.data["fan_speed"]
+
+    @property
+    def humidity(self) -> int:
+        return self.data["humidity"]
+
+    @property
+    def light(self) -> int:
+        return self.data["light"]
+
+    @property
+    def mode(self) -> OperationMode:
+        return OperationMode[OperationModeFanZA5(self.data["mode"]).name]
+
+    @property
+    def power(self) -> bool:
+        return self.data["power"]
+
+    @property
+    def power_off_time(self) -> int:
+        return self.data["power_off_time"]
```

Comment:
Could you do a brief check which properties may be common among the other fans? The plan is to move towards a common, device-type specific APIs, and having matching naming schemes & return types will help when that (breaking) change happens.

Reply:
I have checked the properties of the other fans and adapted some of the new methods to be similar to the existing ones. If you think that additional changes are required, then could you, please, be more specific about which properties you would like me to edit and how?